shiny/ui/_markdown_stream.py

```diff
@@ -0,0 +1,276 @@
+from contextlib import asynccontextmanager
+from typing import AsyncIterable, Iterable, Literal, Union
+
+from htmltools import css
+
+from .. import _utils, reactive
+from .._docstring import add_example
+from .._typing_extensions import TypedDict
+from ..session import require_active_session
+from ..types import NotifyException
+from ..ui.css import CssUnit, as_css_unit
+from . import Tag
+from ._html_deps_py_shiny import markdown_stream_dependency
+
+__all__ = (
+    "markdown_stream_ui",
+    "MarkdownStream",
+)
+
+StreamingContentType = Literal[
+    "markdown",
+    "html",
+    "semi-markdown",
+    "text",
+]
+
+
+class ContentMessage(TypedDict):
+    id: str
+    content: str
+    operation: Literal["append", "replace"]
+
+
+class isStreamingMessage(TypedDict):
+    id: str
+    isStreaming: bool
+
+
+@add_example()
+class MarkdownStream:
+    """
+    Stream markdown (or HTML) content.
+
+    Parameters
+    ----------
+    id
+        A unique identifier for this markdown stream.
```

Comment:
The use of "markdown stream" here gave me pause. I think it should either be a proper noun, like `MarkdownStream`, or it should name more concretely what it identifies. For me, the confusion is around does "markdown stream" refer to the stream of markdown text itself or the UI component where it will be displayed?

Reply:
Done in [d091ebd](https://github.com/posit-dev/py-shiny/pull/1782/commits/d091ebdedb97b6f222f8284c5ac05695d1942b42)